diag_manager2/fms_diag_object.F90

```diff
@@ -378,6 +424,55 @@ function diag_obj_get_varname(obj) result (rslt)
     rslt = obj%varname
 end function diag_obj_get_varname
 
+subroutine diag_obj_set_varname(obj, the_name)
+    class(fms_diag_object), intent(inout) :: obj
+    character(len=*), intent(in) :: the_name
+    obj%varname = the_name
+end subroutine diag_obj_set_varname
+
+!!TODO: Need a data structure other then the 1D axis_ids
+!! Get the axis registration id - per
+function  get_axis_reg_id(this, ai, fi) result (rslt)
+    class(fms_diag_object), intent(in) :: this
```

Comment:
Have i mentioned I hate the use of `this`.  It's the most generic possible name for a derived type variable.